extensions/sql-database-projects/src/test/dialogs/addSqlBindingQuickpick.test.ts

```diff
@@ -77,5 +79,43 @@ describe('Add SQL Binding quick pick', () => {
 		should(spy.calledOnce).be.true('showErrorMessage should have been called exactly once');
 		should(spy.calledWith(errormsg)).be.true(`showErrorMessage not called with expected message '${errormsg}' Actual '${spy.getCall(0).args[0]}'`);
 	});
+
+	it('Should show error connection profile does not connect', async function (): Promise<void> {
+		sinon.stub(utils, 'getAzureFunctionService').resolves(testContext.azureFunctionService.object);
+		sinon.stub(utils, 'getVscodeMssqlApi').resolves(testContext.vscodeMssqlIExtension.object);
+		const spy = sinon.spy(vscode.window, 'showErrorMessage');
+		testContext.azureFunctionService.setup(x => x.getAzureFunctions(TypeMoq.It.isAny())).returns(async () => {
+			return Promise.resolve({
+				success: true,
+				errorMessage: '',
+				azureFunctions: ['af1']
+			});
+		});
+		sinon.stub(azureFunctionUtils, 'getAFProjectContainingFile').resolves(undefined);
+		const errormsg = 'Error connecting to selected connection';
+		testContext.azureFunctionService.setup(x => x.addSqlBinding(TypeMoq.It.isAny(), TypeMoq.It.isAny(), TypeMoq.It.isAny(), TypeMoq.It.isAny(), TypeMoq.It.isAny())).returns(async () => {
```

Comment:
is this test for testing when a connection profile doesn't connect or when addSqlBinding isn't successful? The test name says connection profile not connecting, but this is setting up the addSqlBinding call to return an error message.

Reply:
This is what I was asking you in regards to the setup. This is supposed to be on the connection profile not connecting from my understanding what you wanted me to test but I am not sure why I need to test that when mssql tests that.

Reply:
the reason is to test that the loop logic added here for when a connection isn't successful works correctly